x-pack/plugins/security_solution/server/lib/entity_analytics/risk_engine/get_user_risk_engine_privileges.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { KibanaRequest } from '@kbn/core/server';
+import type { SecurityPluginStart } from '@kbn/security-plugin/server';
+import type { RiskEnginePrivilegesResponse } from './types';
+import {
+  RISK_ENGINE_REQUIRED_ES_CLUSTER_PRIVILEGES,
+  RISK_ENGINE_REQUIRED_ES_INDEX_PRIVILEGES,
+} from '../../../../common/risk_engine';
+
+const groupByPrivilege = (
+  privileges: Array<{
+    privilege: string;
+    authorized: boolean;
+  }>
+): Record<string, boolean> => {
```

Comment:
I think you could make this more specific with a generic for the keys. It might not help much with the type system, but it helps clarify intent IMO: 

```suggestion
const groupPrivilegesByName = <PrivilegeName extends string>(
  privileges: Array<{
    privilege: PrivilegeName;
    authorized: boolean;
  }>
): Record<PrivilegeName, boolean> => {
```

Reply:
I like it, it's not something I have ever thought to do to be honest!